File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];

        /* Generate yearningMap */
        Map<String, Integer> yearningMap = new HashMap<>();
        for(int i = 0; i < name.length; i++){
            yearningMap.put(name[i], yearning[i]);
        }

        /* Compute sum of yearning of photo */
        for(int i = 0; i < answer.length; i++){
            for(String person : photo[i]){
                answer[i] += yearningMap.getOrDefault(person, 0);
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
# 추억 점수: 사진 속의 인물들의 그리움의 합

def solution(names, yearnings, photos):
    # {사람 이름: 그리움}
    dictionary = {name : yearning for name, yearning in zip(names, yearnings)}
    
    # 모든 사진에 대하여
        # 맵에서 찾아서 합 구하기
        
    # 반환
    return [sum(map(lambda x : dictionary[x] if x in dictionary else 0, photo)) for photo in photos]
